src/libs/ReportPrimaryActionUtils.ts

```diff
@@ -0,0 +1,185 @@
+import {
+    isClosedReport,
+    isCurrentUserSubmitter,
+    isExpenseReport,
+    isHoldCreator,
+    isInvoiceReport,
+    isIOUReport,
+    isOpenReport,
+    isPayer,
+    isProcessingReport,
+    isReportApproved,
+} from './ReportUtils';
+import {getSession} from './SessionUtils';
+import {allHavePendingRTERViolation, isDuplicate, isOnHold as isOnHoldTransactionUtils, shouldShowBrokenConnectionViolationForMultipleTransactions} from './TransactionUtils';
+
+function isSubmitAction(report: Report, policy: Policy) {
+    const isExpense = isExpenseReport(report);
+    const isSubmitter = isCurrentUserSubmitter(report.reportID);
+    const isOpen = isOpenReport(report);
+    const isManualSubmitEnabled = getCorrectedAutoReportingFrequency(policy) === CONST.POLICY.AUTO_REPORTING_FREQUENCIES.MANUAL;
+
+    return isExpense && isSubmitter && isOpen && isManualSubmitEnabled;
+}
+
+function isApproveAction(report: Report, policy: Policy, reportTransactions: Transaction[]) {
+    const isExpense = isExpenseReport(report);
+    const isApprover = isApprovedMember(policy, getCurrentUserAccountID());
+    const isApprovalEnabled = policy.approvalMode && policy.approvalMode !== CONST.POLICY.APPROVAL_MODE.OPTIONAL;
+
+    if (!isExpense || !isApprover || !isApprovalEnabled) {
+        return false;
+    }
+
+    const isOneExpenseReport = isExpense && reportTransactions.length === 1;
+    const isOnHold = reportTransactions.some(isOnHoldTransactionUtils);
+    const isProcessing = isProcessingReport(report);
+    const isOneExpenseReportOnHold = isOneExpenseReport && isOnHold;
+
+    if (isProcessing || isOneExpenseReportOnHold) {
+        return true;
+    }
+
+    return false;
+}
+
+function isPayAction(report: Report, policy: Policy) {
+    const isExpense = isExpenseReport(report);
+    const isReportPayer = isPayer(getSession(), report, false, policy);
+    const isPaymentsEnabled = arePaymentsEnabled(policy);
+    const isApproved = isReportApproved({report});
+    const isClosed = isClosedReport(report);
+    const isFinished = isApproved || isClosed;
+
+    if (isReportPayer && isExpense && isPaymentsEnabled && isFinished) {
+        return true;
+    }
+
+    const isProcessing = isProcessingReport(report);
+    const isInvoice = isInvoiceReport(report);
+    const isIOU = isIOUReport(report);
+
+    if ((isInvoice || isIOU) && isProcessing) {
+        return true;
+    }
+
+    return false;
+}
+
+function isExportAction(report: Report, policy: Policy) {
+    const isExporter = isPrefferedExporter(policy);
+    if (!isExporter) {
+        return false;
+    }
+
+    const syncEnabled = isAutoSyncEnabled(policy);
+    if (syncEnabled) {
+        return false;
+    }
+
+    const hasAccountingConnection = hasAccountingConnections(policy);
+    if (!hasAccountingConnection) {
+        return false;
+    }
+
+    const isReimbursed = report.statusNum === CONST.REPORT.STATUS_NUM.REIMBURSED;
```

Comment:
We should re-use [isSettled](https://github.com/Expensify/App/blob/7d0bcb4946fbdd1b6b57a285f8678ca80d94d07a/src/libs/ReportUtils.ts#L1256-L1276) or add similar logic here to account for `isWaitingOnBankAccount`
```suggestion
    const isReimbursed = report.statusNum === CONST.REPORT.STATUS_NUM.REIMBURSED;
```

Reply:
By the way there is contradiction in `isSettled` function in those two ifs: https://github.com/Expensify/App/blob/7d0bcb4946fbdd1b6b57a285f8678ca80d94d07a/src/libs/ReportUtils.ts#L1265-L1273

Reply:
lol nice catch. I think we need to update that. We should change `if (isEmptyObject(report) || report.isWaitingOnBankAccount)` to `if (isEmptyObject(report))`